Mathlib/RingTheory/TensorProduct/DirectLimit/FG.lean

```diff
@@ -0,0 +1,375 @@
+
+* `DirectedSystem.rTensor`: given a directed system `M i`,
+the directed system of modules `M i ⊗[R] N`.
+
+* `rTensor_fgEquiv`: a tensor product `M ⊗[R] N` is the direct limit
+of the tensor products `P ⊗[R] N`, for finitely generated submodules `P` of `M`.
+
+* `DirectedSystem.lTensor`: given a directed system `N i`,
+the directed system of modules `M ⊗[R] N i`.
+
+* `lTensor_fgEquiv`: a tensor product `M ⊗[R] N` is the direct limit
+of the tensor products `M ⊗[R] Q`, for finitely generated submodules `Q` of `N`.
+
+* `TensorProduct.exists_rTensor_of_fg`: any element `t:  M ⊗[R] N`  can be lifted
+to some `P ⊗[R] N`, for fome finitely generated submodule `P` of `M`.
+
+* `TensorProduct.eq_of_fg_of_subtype_eq`;
+given a finitely generated submodule `P` of `M` and `t, t' : P ⊗[R] N`
+which have the same image in `M ⊗[R] N`, there exists a finitely generated submodule `Q` of `M`
+which contains `P` such that `t` and `t'` have the same image in `Q ⊗[R] N`.
+
+* `TensorProduct.eq_of_fg_of_subtype_eq₂`;
+given finitely generated submodules `P` and `P'`of `M`, `t : P ⊗[R] N` and `t' : P' ⊗[R] N`
+which have the same image in `M ⊗[R] N`, there exists a finitely generated submodule `Q` of `M`
+which contains `P` and `P'` and such that `t` and `t'` have the same image in `Q ⊗[R] N`.
+
+* `TensorProduct.Algebra.exists_of_fg`: any element `t : S ⊗[R] N` can be lifted
+to `A ⊗[R] N`, for some finitely generated subalgebra `A` of `S`.
+
+* `TensorProduct.Algebra.eq_of_fg_of_subtype_eq`:
+given a finitely generated subalgebra `A` of `S`, `t : A ⊗[R] N` and `t' : A ⊗[R] N`
+which have the same image in `S ⊗[R] N`, there exists a finitely generated subalgebra `B` of `S`
+which contains `A` and such that `t` and `t'` have the same image in `B ⊗[R] N`.
+
+* `TensorProduct.Algebra.eq_of_fg_of_subtype_eq₂`:
+given finitely generated subalgebras `A` and `A'`of `S`, `t : A ⊗[R] N` and `t' : A' ⊗[R] N`
+which have the same image in `S ⊗[R] N`, there exists a finitely generated subalgebra `B` of `S`
+which contains `A` and `A'` and such that `t` and `t'` have the same image in `B ⊗[R] N`.
+
+* `Submodule.exists_fg_of_baseChange_eq_zero`: if `t : S ⊗[R] M`
+mapst to `0` in `S ⊗[R] N`, there exists a finitely generated subalgebra `A` of `S`
+and `u : A ⊗[R] M` which maps to `t` in `S ⊗[R] M`, and to `0` in `S ⊗[R] N`.
+
+## TODO
+
+* The results are valid in the context of `AddCommMonoid M` over a `Semiring`.
+However,  tensor products in mathlib require commutativity of the scalars,
+and direct limits of modules are restricted to modules over rings.
+
+-/
+
+open Submodule LinearMap
+
+section Semiring
+
+universe u v
+variable {R : Type u} [Semiring R] {M : Type*} [AddCommMonoid M] [Module R M]
+
+/-- The directed system of finitely generated submodules of `M` -/
+theorem DirectedSystem.Submodule_fg :
+    DirectedSystem (ι := {P : Submodule R M // P.FG}) (F := fun P ↦ P.val)
+    (f := fun ⦃P Q⦄ (h : P ≤ Q) ↦ Submodule.inclusion h) where
```

Comment:
Is this not the same as [Module.fgSystem](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Colimit/Finiteness.html#Module.fgSystem)?